pkg/xray/resource_xray_workers_count.go

```diff
@@ -0,0 +1,280 @@
+		NewContent
+		Existing int `json:"existing_content"`
+	}
+
+	/*
+		API returns the follow JSON structure:
+		{
+		    "index": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "persist": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "analysis": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "alert": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "impact_analysis": {
+		        "new_content": 2
+		    },
+		    "notification": {
+		        "new_content": 2
+		    }
+		}
+	*/
+	type WorkersCount struct {
+		Index          NewExistingContent `json:"index"`
+		Persist        NewExistingContent `json:"persist"`
+		Analysis       NewExistingContent `json:"analysis"`
+		Alert          NewExistingContent `json:"alert"`
+		ImpactAnalysis NewContent         `json:"impact_analysis"`
+		Notification   NewContent         `json:"notification"`
+	}
+
+	var newHclContentConstructor = func(src interface{}) map[string]interface{} {
+		return map[string]interface{}{
+			"new_content": src.(NewContent).New,
+		}
+	}
+
+	var newExistingHclContentConstructor = func(src interface{}) map[string]interface{} {
+		return map[string]interface{}{
+			"new_content":      src.(NewExistingContent).New,
+			"existing_content": src.(NewExistingContent).Existing,
+		}
+	}
+
+	var packContent = func(d *schema.ResourceData, attr string, src interface{}, hclContentConstructor func(src interface{}) map[string]interface{}) []error {
+		setValue := util.MkLens(d)
+
+		resource := workersCountSchema[attr].Elem.(*schema.Resource)
+		content := hclContentConstructor(src)
+		return setValue(attr, schema.NewSet(schema.HashResource(resource), []interface{}{content}))
+	}
+
+	var packWorkersCount = func(d *schema.ResourceData, workersCount WorkersCount) diag.Diagnostics {
+		var errors []error
+
+		errors = append(errors, packContent(d, "index", workersCount.Index, newExistingHclContentConstructor)...)
+		errors = append(errors, packContent(d, "persist", workersCount.Persist, newExistingHclContentConstructor)...)
+		errors = append(errors, packContent(d, "analysis", workersCount.Analysis, newExistingHclContentConstructor)...)
+		errors = append(errors, packContent(d, "alert", workersCount.Alert, newExistingHclContentConstructor)...)
+		errors = append(errors, packContent(d, "impact_analysis", workersCount.ImpactAnalysis, newHclContentConstructor)...)
+		errors = append(errors, packContent(d, "notification", workersCount.Notification, newHclContentConstructor)...)
+
+		if len(errors) > 0 {
+			return diag.Errorf("failed to pack workers count %q", errors)
+		}
+
+		return nil
+	}
+
+	var unpackContent = func(d *schema.ResourceData, attr string, constructor func(map[string]interface{}) interface{}) interface{} {
+		var content interface{}
+		if v, ok := d.GetOk(attr); ok {
+			s := v.(*schema.Set).List()[0].(map[string]interface{})
```

Comment:
Which part is weird? `v.(*schema.Set).List()` returns `[]interface{}`. Since we limit this attribute to `minItems = 1`, and `maxItems = 1`, we can pick the first item and convert it to `map[string]interface{}`.

Reply:
Can you put that explanation in a comment - It wasn't obvious to me